app/EdgeCollapse/ShortestEdgeCollapse.cpp

```diff
@@ -0,0 +1,59 @@
+#include <wmtk/TriMesh.h>
+#include <wmtk/utils/VectorUtils.h>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include "EdgeCollapse.h"
+
+using namespace Edge2d;
+// delete the old two vert position, add the new at the middle of the old two points
+void Edge2d::EdgeCollapse::update_position(size_t v1, size_t v2, Tuple& new_vert)
+{
+    size_t new_vid = new_vert.get_vid();
```

Comment:
you can directly use `std::min(v1,v2)` as the new_vid. This avoid resizing the array or traversing the array all the time.